### PR TITLE
Profile Test Routes Completed

### PR DESCRIPTION
**Pull request by RajendraBhagroo**
Next Steps: Remove Requirements From Profile Model And Enforce Them In Validation Files. All While Still Passing Tests [Update Profile Tests To Reflect Cases In Profile Validation]

### CODE REVIEW

File: server/v1/tests/profile.test.js [hunk 1]
```diff
@@ -21,3 +104,398 @@ describe("GET /api/v1/profile/test", () => {
       );
   });
 });
+
+/*
+ * @route   GET /api/v1/profile
+ * @desc    Return Current Users Profile
```
Comment (gint0kix) [inline server/v1/tests/profile.test.js:110]: Description doesn't match whats being done

File: server/v1/tests/profile.test.js [hunk 2]
```diff
@@ -21,3 +104,398 @@ describe("GET /api/v1/profile/test", () => {
...
+      .get("/api/v1/profile/handle/TaylorSwift")
+      .set("Accept", "application/json")
+      .expect(200, done);
+  });
+});
+
+/*
+ * @route       GET /api/v1/profile/handle/:handle
+ * @params URL  {handle}
+ * @desc        Get Profile By Handle
```
Comment (gint0kix) [inline server/v1/tests/profile.test.js:216]: Description needs to be fixed, to reflect the test

File: server/v1/tests/profile.test.js [hunk 3]
```diff
@@ -21,3 +104,398 @@ describe("GET /api/v1/profile/test", () => {
...
+      .get(`/api/v1/profile/user/${user_id}`)
+      .set("Accept", "application/json")
+      .expect(200, done);
+  });
+});
+
+/*
+ * @route       GET /api/v1/profile/user/:user_id
+ * @params URL  {user_id}
+ * @desc        Get Profile By User ID
```
Comment (gint0kix) [inline server/v1/tests/profile.test.js:246]: Description needs to be fixed, to reflect the test

File: server/v1/tests/profile.test.js [hunk 4]
```diff
@@ -21,3 +104,398 @@ describe("GET /api/v1/profile/test", () => {
...
+      .set("Authorization", token)
+      .set("Accept", "application/json")
+      .expect(200, done);
+  });
+});
+
+/*
+ * @route       DELETE /api/v1/profile/education/:edu_id
+ * @params URL  {edu_id}
+ * @desc        Delete Education From Profile
```
Comment (gint0kix) [inline server/v1/tests/profile.test.js:399]: Description needs to be fixed, to reflect the test

File: server/v1/tests/profile.test.js [hunk 5]
```diff
@@ -21,3 +104,398 @@ describe("GET /api/v1/profile/test", () => {
...
+      .set("Authorization", token)
+      .set("Accept", "application/json")
+      .expect(200, done);
+  });
+});
+
+/*
+ * @route       DELETE /api/v1/profile/experience/:exp_id
+ * @params URL  {exp_id}
+ * @desc        Delete Experience From Profile
```
Comment (gint0kix) [inline server/v1/tests/profile.test.js:430]: Description needs to be fixed, to reflect the test

File: server/v1/tests/profile.test.js [hunk 6]
```diff
@@ -21,3 +104,398 @@ describe("GET /api/v1/profile/test", () => {
...
+      .delete(`/api/v1/profile/coursesTeaching/${coursesTeaching_id}`)
+      .set("Authorization", token)
+      .set("Accept", "application/json")
+      .expect(200, done);
+  });
+});
+
+/*
+ * @route   DELETE /api/v1/profile
+ * @desc    Delete User And Profile
```
Comment (gint0kix) [inline server/v1/tests/profile.test.js:476]: Description needs to be fixed, to reflect the test

**Review by gint0kix — CHANGES_REQUESTED** [review]
Just some errors in the descriptions that need to be fixed

File: server/v1/tests/profile.test.js (hunk 1 above)
Comment (RajendraBhagroo) [reply server/v1/tests/profile.test.js:110]: I left original description of route so developers know what it is supposed to do, and in the "describe" I describe how I'm not using as intended and it is expected to fail

**Comment by RajendraBhagroo** [comment]
@gint0kix  Im happy you looked over the file!

Those descriptions are the original descriptions for the route and I intended to always keep the original description so developers know the intended functionality of the route. In the test under the route description I meantiom that it is expected to fail if I intended the route to fail. I found it helpful to know the original intention of the route to help me make unit tests to implicitly fail. Thank you for the review, keep that eye sharp. In every route in the testing file I kept the original descriptions, not just the ones that fail and you picked up on that 👍🏾. 

Users.test.js follows another convention where the original description is modified and that's a great talking point for potential employers as to why it isn't consistent. Technically you stayed consistent since you wrote users tests file and I was going to follow description but profile had a tedious amount of tests to write and I was jumping between files and code so often I kept the description as is only so I didn't have to keep referring to original route to make the test. 

In circleCI we have a whopping 32 tests ensuring our API doesn't break unintentionally and it's caught me a few times already. Tests are great!

**Review by gint0kix — APPROVED** [review]
Approved

**Review by Gold-Turtle — APPROVED** [review]
Approved

**Pull request merged**